### PR TITLE
bazel/ci: Shift GCS upload script to bazel

### PR DESCRIPTION
**Pull request by phlax**
Commit Message:
Additional Description:
Risk Level:
Testing:
Docs Changes:
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Fixes commit #PR or SHA]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

File: tools/gcs/upload.sh [hunk 1]
```diff
@@ -40,32 +60,31 @@ fi
...
-      //tools/gsutil \
-      -- -mq rsync \
-         -dr "${SOURCE_DIRECTORY}" \
+"${GSUTIL}" \
+         -mq rsync \
+         -dr "${UPLOAD_DIRECTORY}" \
          "gs://${GCS_LOCATION}"
 
+if [[ -z "${REDIRECT_PATH}" ]]; then
+    echo "Artifacts uploaded to: https://storage.googleapis.com/${GCS_LOCATION}" >&2
```
Comment (adisuissa) [inline tools/gcs/upload.sh:73]: doesn't this need `/index.html` after the GCS_location (similar to the link below)?
I'm guessing that the default will just redirect/show the index.html page, but just want to make sure.

**Review by adisuissa — APPROVED** [review]
LGTM, just left minor comment.

File: tools/gcs/upload.sh (hunk 1 above)
Comment (phlax) [reply tools/gcs/upload.sh:73]: no - in this case the index.html doesnt exist (or at least most likely doesnt), so would not work

that said, the links here dont work 100% as expected - in this PR i just wanted to move the code and keep existing behaviour - the reason being, that most of the usage can be just removed in the next PR i want to land, and i figured i can clean it up after that

**Pull request merged**